### PR TITLE
arch/risc-v: Apply common mtime driver to mtime based chps

### PR DESCRIPTION
**Pull request by no1wudi**
## Summary
See #6020, apply it to k210/bl602/qemu_rv etc
## Impact
risc-v based chips
## Testing
rv-virt/k210/bl602

### CODE REVIEW

**Comment by xiaoxiang781216** [comment]
@pussuw could you try this patch with MPFS?

**Comment by pussuw** [comment]
> @pussuw could you try this patch with MPFS?

Yes, goes to tomorrow though

File: arch/risc-v/src/c906/c906_timerisr.c [hunk 1]
```diff
@@ -117,15 +64,7 @@ static int c906_timerisr(int irq, void *context, void *arg)
...
-  irq_attach(RISCV_IRQ_MTIMER, c906_timerisr, NULL);
-
-  /* Reload CLINT mtimecmp */
-
-  c906_reload_mtimecmp();
-
-  /* And enable the timer interrupt */
-
-  up_enable_irq(RISCV_IRQ_MTIMER);
+  up_alarm_set_lowerhalf(riscv_mtimer_initialize(
```
Comment (pkarashchenko) [inline arch/risc-v/src/c906/c906_timerisr.c:67]: can we use temp variable here, just for easy reading?

**Comment by pkarashchenko** [comment]
In general LGTM!

**Comment by pkarashchenko** [comment]
I still observe some warnings on a CI
```
.config:118:warning: symbol value '' invalid for ARCH_CPU_COUNT
  Building NuttX...
Makefile:123: target 'riscv_cpuindex.o' given more than once in the same rule
Makefile:123: target 'riscv_cpuindex.o' given more than once in the same rule
Makefile:123: target 'riscv_cpuindex.o' given more than once in the same rule
Makefile:123: target 'riscv_cpuindex.o' given more than once in the same rule
Makefile:123: target 'riscv_cpuindex.o' given more than once in the same rule
  Normalize maix-bit/smp2
.config:118:warning: symbol value '' invalid for ARCH_CPU_COUNT
.config:118:warning: symbol value '' invalid for ARCH_CPU_COUNT
```
I think that `.config:118:warning: symbol value '' invalid for ARCH_CPU_COUNT` is known and should be addressed soon, but `Makefile:123: target 'riscv_cpuindex.o' given more than once in the same rule` seems to be a new one

**Comment by xiaoxiang781216** [comment]
> I still observe some warnings on a CI
> 
> ```
> .config:118:warning: symbol value '' invalid for ARCH_CPU_COUNT
>   Building NuttX...
> Makefile:123: target 'riscv_cpuindex.o' given more than once in the same rule
> Makefile:123: target 'riscv_cpuindex.o' given more than once in the same rule
> Makefile:123: target 'riscv_cpuindex.o' given more than once in the same rule
> Makefile:123: target 'riscv_cpuindex.o' given more than once in the same rule
> Makefile:123: target 'riscv_cpuindex.o' given more than once in the same rule
>   Normalize maix-bit/smp2
> .config:118:warning: symbol value '' invalid for ARCH_CPU_COUNT
> .config:118:warning: symbol value '' invalid for ARCH_CPU_COUNT
> ```
> 
> I think that `.config:118:warning: symbol value '' invalid for ARCH_CPU_COUNT` is known and should be addressed soon, but `Makefile:123: target 'riscv_cpuindex.o' given more than once in the same rule` seems to be a new one

Both issues are fixed by https://github.com/apache/incubator-nuttx/pull/5985

**Review by pussuw — APPROVED** [review]
LGTM

**Review by pkarashchenko — APPROVED** [review]

File: arch/risc-v/src/c906/c906_timerisr.c (hunk 1 above)
Comment (no1wudi) [reply arch/risc-v/src/c906/c906_timerisr.c]: OK

**Pull request merged**